### PR TITLE
community[minor]:  [GoogleApiYoutubeLoader] Replace API used in _get_document_for_channel from search to playlistItem

### PR DESCRIPTION
**Pull request by TupleType**
- **Description:** Search has a limit of 500 results, playlistItems doesn't. Added a class in except clause to catch another common error.
- **Issue:** None
- **Dependencies:** None
- **Twitter handle:** @TupleType

### CODE REVIEW

File: libs/community/langchain_community/document_loaders/youtube.py [hunk 1]
```diff
@@ -7,6 +7,7 @@
 from pathlib import Path
 from typing import Any, Dict, Generator, List, Optional, Sequence, Union
 from urllib.parse import parse_qs, urlparse
+from xml.etree.ElementTree import ParseError
```
Comment (eyurtsev) [inline libs/community/langchain_community/document_loaders/youtube.py:10]: You can do this:

```python
from xml.etree.ElementTree import ParseError # OK: user-must-opt-in
```

The linter is flagging the fact that the underlying library is relying on the built-in XML library.


It's essentially surfacing this:

https://github.com/jdepoix/youtube-transcript-api/blob/master/youtube_transcript_api/_transcripts.py#L10

Depending on user environment the built in may or may not have vulnerabilities:
https://docs.python.org/3/library/xml.html#xml-vulnerabilities

Do you know if the underlying library reading from a google provided XML file? If so, while not amazing this is probably fine.

Could you add a security note to the GoogleAPIClient saying that it relies on the standard xml library, but we're viewing the input as trusted in this case?

Comment (TupleType) [reply libs/community/langchain_community/document_loaders/youtube.py:10]: I don' know the library well enough to determine that but this is the error i'm getting for https://www.youtube.com/watch?v=pM8e0Dbzopk:
File "C:\Users\user\repos\rag-demo\loader.py", line 71, in _get_document_for_playlist
    transcript = self._get_transcripe_for_video_id(
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\repos\rag-demo\.venv\Lib\site-packages\langchain_community\document_loaders\youtube.py", line 410, in _get_transcripe_for_video_id
    transcript_pieces = transcript.fetch()
                        ^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\repos\rag-demo\.venv\Lib\site-packages\youtube_transcript_api\_transcripts.py", line 292, in fetch
    return _TranscriptParser(preserve_formatting=preserve_formatting).parse(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\repos\rag-demo\.venv\Lib\site-packages\youtube_transcript_api\_transcripts.py", line 358, in parse
    for xml_element in ElementTree.fromstring(plain_data)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\AppData\Local\Programs\Python\Python311\Lib\xml\etree\ElementTree.py", line 1338, in XML
    parser.feed(text)
xml.etree.ElementTree.ParseError: not well-formed (invalid token): line 57, column 99

**Comment by TupleType** [comment]
dependency checks failed but it doesn't have logs

File: libs/community/langchain_community/document_loaders/youtube.py [hunk 2]
```diff
@@ -7,6 +7,7 @@
 from pathlib import Path
 from typing import Any, Dict, Generator, List, Optional, Sequence, Union
 from urllib.parse import parse_qs, urlparse
+from xml.etree.ElementTree import ParseError  # OK: trusted-source
```
Comment (eyurtsev) [inline libs/community/langchain_community/document_loaders/youtube.py:10]: Changed the comment here and the security note style

**Review by eyurtsev — APPROVED** [review]

File: libs/community/langchain_community/document_loaders/youtube.py [hunk 3]
```diff
@@ -452,34 +463,32 @@ def _get_document_for_channel(self, channel: str, **kwargs: Any) -> List[Documen
             )
 
         channel_id = self._get_channel_id(channel)
-        request = self.youtube_client.search().list(
+        uploads_playlist_id = self._get_uploads_playlist_id(channel_id)
```
[the remark this answers is not in the document]
Comment (TupleType) [reply libs/community/langchain_community/document_loaders/youtube.py:466]: yes it's backward compatible

**Pull request merged**